### ISSUE TITLE
placeholder attribute not exposed

### ISSUE THREAD

**Issue by adamschnaare** [vaadin/vaadin-combo-box#170]
The `placeholder` attribute for the input is not exposed.

**Comment by Saulis** [comment]
Hi there and thanks for your interest in `vaadin-combo-box` !

Is there a specific reason you need the `placeholder` attribute to be set? If you need to display a hint description about the expected value for the field, you can use the `label` attribute.

```<vaadin-combo-box label="First Name"></vaadin-combo-box>```

**Comment by adamschnaare** [comment]
It might not be the paper standard way to do it, but it feels like since it
is a standard HTML option, it should be in there. My use case specifically
didn't require the label, but just the placeholder text.

On Thu, Feb 25, 2016 at 3:57 PM, Sauli Tähkäpää <notifications@github.com>
wrote:

> Hi there and thanks for your interest in vaadin-combo-box !
>
> Is there a specific reason you need the placeholder attribute to be set?
> If you need to display a hint description about the expected value for the
> field, you can use the label attribute.
>
> <vaadin-combo-box label="First Name"></vaadin-combo-box>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vaadin/vaadin-combo-box/issues/170#issuecomment-189004031>
> .
>



-- 

Sincerely,

~~~~~~~~~~~~~~~~~~~~~~
*Adam Schnaare*
UX Designer/Engineer
Worship Director
715.615.2832

**Comment by Saulis** [comment]
Right, so was the problem for you that the label was floating on top of the input? Would it work for you if you could stop from floating? See, #107 

The `placeholder` is a bit problematic because it will be obscured if used at the same time with `label`,  and it would also be an overlapping feature. Any comments @jounik ?

**Comment by adamschnaare** [comment]
The floating label is fine, but a different use case. And I believe they are solving two different UX problems. The floating label identifies what the field is, and a placeholder would be text helping the user know and example of what data to input.

**For example:**
Label: `State`
Placeholder: `California, New Mexico, etc...`

I think both should be included in any complete input solution, allowing developers and ux-ers to use one or both depending on their inclination.

**Comment by Saulis** [comment]
We've added #198 to add a way for users to provide their own custom input elements.

**Comment by jouni** [comment]
@adamschnaare, we’ve started designing and prototyping #198 more closely, so now would be a great time to start giving more feedback how it’s going to look like. Check out the comments in that issue and let’s continue the discussion there. Thanks!

**Comment by Saulis** [comment]
Fixed in #254
